src/wobbly-effect.cpp

```diff
@@ -0,0 +1,570 @@
+     * paint box when we're just considering the
+     * actor alone */
+    RAIIActorPaintBox forced_client_paint_box (effect_class);
+    ClutterActorBox   rect;
+
+    /* If clutter_actor_get_paint_box fails
+     * we should fall back to the actor size at this point */
+    if (clutter_actor_get_paint_box (actor, &rect))
+        clutter_actor_box_get_size (&rect, width, height);
+    else
+        clutter_actor_get_size (actor, width, height);
+}
+
+static void
+endless_shell_fx_wobbly_size_changed (GObject    *object,
+                                      GParamSpec *,
+                                      gpointer   user_data)
+{
+    ClutterActor        *actor = CLUTTER_ACTOR (object);
+    EndlessShellFXWobbly *effect = ENDLESS_SHELL_FX_WOBBLY (user_data);
+    EndlessShellFXWobblyPrivate *priv =
+        reinterpret_cast <EndlessShellFXWobblyPrivate *> (endless_shell_fx_wobbly_get_instance_private (effect));
+
+    /* We don't ensure a timeline here because we only want to redistribute
+     * non-anchor points if we're already grabbed, which the wobbly effect will
+     * do internally anyways */
+    if (priv->model) {
+        float actor_width, actor_height;
+        endless_shell_fx_get_actor_only_paint_box_size (effect,
+                                                        actor,
+                                                        &actor_width,
+                                                        &actor_height);
+
+        /* If we have any pending anchors, we should release them now -
+         * the model move and resize code explicitly does not move
+         * anchors around (because that'd put them out of sync with
+         * the cursor) */
+        remove_anchor_if_pending (priv);
+
+        priv->model->ResizeModel (actor_width, actor_height);
+        priv->model->MoveModelTo (wobbly::Point (0, 0));
+
+    }
+}
+
+static void
+endless_shell_fx_wobbly_set_actor (ClutterActorMeta *actor_meta,
+                                   ClutterActor     *actor)
+{
+    CLUTTER_ACTOR_META_CLASS (endless_shell_fx_wobbly_parent_class)->set_actor (actor_meta,
+                                                                                 actor);
+
+    EndlessShellFXWobbly *wobbly_effect = ENDLESS_SHELL_FX_WOBBLY (actor_meta);
+    EndlessShellFXWobblyPrivate *priv =
+        reinterpret_cast <EndlessShellFXWobblyPrivate *> (endless_shell_fx_wobbly_get_instance_private (wobbly_effect));
+
+    /* If we're grabbed, we have to immediately get rid of the grab
+     * as the grab depends on objects inside the model */
+    if (priv->anchor)
+    {
+        delete priv->anchor;
+        priv->anchor = nullptr;
+        priv->ungrab_pending = false;
+    }
+
+    if (priv->model)
+    {
+        delete priv->model;
+        priv->model = nullptr;
+    }
+
+    if (priv->timeout_id)
+    {
+        g_source_remove (priv->timeout_id);
+        priv->timeout_id = 0;
+    }
+
+    if (priv->actor)
```

Comment:
You could avoid storing the reference to the actor here entirely, since ClutterActorMeta will already keep a reference as long as the actor is alive.
In this case, you'd access the old reference before chaining up to parent in this function.